aptos-move/framework/aptos-framework/sources/coin.move

```diff
@@ -139,6 +153,59 @@ module aptos_framework::coin {
         *allow_upgrades = allowed;
     }
 
+    //
+    //  Aggregator coin functions
+    //
+
+    /// Creates a new aggregatable coin with value overflowing on `limit`. Note that this function can
+    /// only be called by Aptos Framework (0x1) account for now becuase of `create_aggregator`.
+    public fun initialize_aggregator_coin<CoinType>(aptos_framework: &signer): AggregatorCoin<CoinType> {
+        let aggregator = aggregator_factory::create_aggregator(aptos_framework, MAX_U64);
+        AggregatorCoin<CoinType> {
+            value: aggregator,
+        }
+    }
+
+    /// Returns true if the value of aggregatable coin is zero.
+    public fun is_zero<CoinType>(coin: &AggregatorCoin<CoinType>): bool {
+        let amount = aggregator::read(&coin.value);
+        amount == 0
+    }
+
+    /// Drains the aggregatable coin, setting it to zero and returning a standard coin.
+    public fun drain<CoinType>(coin: &mut AggregatorCoin<CoinType>): Coin<CoinType> {
+        let amount = aggregator::read(&coin.value);
+        assert!(amount <= MAX_U64, error::out_of_range(EAGGREGATOR_COIN_VALUE_TOO_LARGE));
+
+        aggregator::sub(&mut coin.value, amount);
+        Coin<CoinType> {
+            value: (amount as u64),
+        }
+    }
+
+    /// Aggregates `coin` into aggregatable coin (`dst_coin`).
+    public fun aggregate<CoinType>(dst_coin: &mut AggregatorCoin<CoinType>, coin: Coin<CoinType>) {
+        let Coin { value } = coin;
+        let amount = (value as u128);
+        aggregator::add(&mut dst_coin.value, amount);
+    }
+
+    /// Collects a specified amount form an account to aggregatable coin.
+    public fun collect_from<CoinType>(
```

Comment:
seems dangerous to have it as public fun and only requires address, imagine future we enable 3rd party use of the aggregatable coin

Reply:
Indeed, what about CollectCapability? Otherwise if we make it friend or take signer -- might be difficult to upgrade

Reply:
It seems like making this friend would create a circular dependency at the moment. Let me check if I can change to make this a friend function.